pkg/services/ngalert/store/alert_rule.go

```diff
@@ -359,10 +369,20 @@ func (st DBstore) GetNamespaceAlertRules(query *ngmodels.ListNamespaceAlertRules
 // GetRuleGroupAlertRules is a handler for retrieving rule group alert rules of specific organisation.
 func (st DBstore) GetRuleGroupAlertRules(query *ngmodels.ListRuleGroupAlertRulesQuery) error {
 	return st.SQLStore.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
-		alertRules := make([]*ngmodels.AlertRule, 0)
-
 		q := "SELECT * FROM alert_rule WHERE org_id = ? and namespace_uid = ? and rule_group = ?"
-		if err := sess.SQL(q, query.OrgID, query.NamespaceUID, query.RuleGroup).Find(&alertRules); err != nil {
+		args := []interface{}{query.OrgID, query.NamespaceUID, query.RuleGroup}
+
+		if query.DashboardUID != "" {
```

Comment:
I'm not sure if this is part of the requirements or not, but are you only able to filter by panelID if you have specified a dashboard? My guess here is that it is given you even have a test case for it `	// Now, let's check we get that panel_id without dashboard_uid does not filter rules`.

But want to double-check for the sake of my understanding? /cc @domasx2

Reply:
That's correct. The thinking here was:

1. When would we want to get the rules for a panel without the dashboard?
2. We would need to create a second database index that could be used in queries which do not have a where condition on the `dashboard_uid` column as the current index is designed for queries that have where conditions on the `org_id`, `namespace_id`, `dashboard_uid` and `panel_id` in that order.

Reply:
given that the `panel_id` is a number, there is no much point filtering only by the `panel_id`, we will always need the `dashboard_id` as well.

Reply:
There is a use case to get only panel rules. But there is no situation where frontend knows panel id, but does not know the dasbhoard id. So it would be ok to make dashbaord id mandatory if panel id is provided, if it makes anything easier

Reply:
Thanks Domas for that context - we'll make sure to note it down for the future.